Tasks/AzureRmWebAppDeployment/operations/KuduServiceUtility.ts

```diff
@@ -127,6 +127,56 @@ export class KuduServiceUtility {
         }
     }
 
+    public async zipDeploy(packagePath: string, appOffline?: boolean, customMessage?: any): Promise<void> {
+        try {
+            
+            if(tl.stats(packagePath).isDirectory()) {
+                let tempPackagePath = deployUtility.generateTemporaryFolderOrZipPath(tl.getVariable('AGENT.TEMPDIRECTORY'), false);
+                packagePath = await zipUtility.archiveFolder(packagePath, "", tempPackagePath);
+                tl.debug("Compressed folder " + packagePath + " into zip : " +  packagePath);
+            }
+
+            if(appOffline) {
+                await this._appOfflineKuduService(physicalRootPath, true);
+                tl.debug('Wait for 10 seconds for app_offline to take effect');
+                await webClient.sleepFor(10);
+            }
+
+            var deploymentRequestDetails = this._getUpdateHistoryRequest(true, null, customMessage);
+            var queryParameters: Array<string> = [
+                'isAsync=true',
+                'message=' + encodeURIComponent(deploymentRequestDetails.message),
+                'deployer=' + 'VSTS',
+                'author=' + deploymentRequestDetails.author,
+                'details=' + encodeURIComponent(deploymentRequestDetails.details)
+            ];
+
+            var deploymentDetails = await this._appServiceKuduService.zipDeploy(packagePath, queryParameters);
+
+            try {
+                var logs = await this._appServiceKuduService.getFileContent(`/site/deployments/${deploymentDetails.id}`, 'log.log');
```

Comment:
How big can this be?

Reply:
Not sure about the size, it depends on the number of files and folders present within the package.
Also, we are not failing here, when we fail to get the logs, as we are primarily interested on the status first. We can also point the API to the user, where he could get the deployment logs, in the catch part.